addons_neokingdom/neok_hr_expose_for_website/controllers/main.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+import logging
+import json
+
+from odoo import http, _
+from odoo.http import request, Response
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteHrRecruitment(http.Controller):
+    @http.route('/neok-api/positions', auth="public", csrf=False, methods=['GET'], type='json')
+    def neok_api_positions(self):
+        try:
+            job_ids = request.env['hr.job'].sudo().search([('website_published', '=', True)], order="sequence, create_date desc")
+
+            return [{
+                'id': _job.id,
+                'title': _job.name or '',
+                'department': _job.department_id.name or '',
+                'location': _job.address_id.name or '',
+                'contract_type': _job.contract_type_id.name or '',
+                'recruiter': _job.user_id.name or ''
+            } for _job in job_ids]
+        except Exception as e:
+            _logger.error('Error while fetching positions: %s', str(e))
+            return Response(
+                json.dumps({'msg': 'Error while fetching positions. Please try again later.'}),
+                status=500,
+                mimetype='application/json'
+            )
+
+    @http.route('/neok-api/position/details', auth="public", csrf=False, methods=['POST'], type='json')
+    def neok_api_position_details(self):
+        try:
+            _data = json.loads(request.httprequest.data)
+            job_id = _data.get('job_id', False)
+
+            if not job_id:
+                return Response(
+                    json.dumps({'msg': 'Position not found.'}),
+                    status=404,
+                    mimetype='application/json'
+                )
+
+            job_id = request.env['hr.job'].sudo().browse(int(job_id))
+
+            if not job_id.exists():
+                return Response(
+                    json.dumps({'msg': 'Position not found.'}),
+                    status=404,
+                    mimetype='application/json'
+                )
+            if not job_id.website_published:
+                return Response(
+                    json.dumps({'msg': 'Position is closed!'}),
+                    status=400,
+                    mimetype='application/json'
+                )
```

Comment:
Use a method for "message, code" to return the response ... and only use one return statement for the exception case.

_check_error_message e.g.

Reply:
it saves 2 lines but doesnt reduce to much repetition, since its just creating an object. but im fine with it.